### PR TITLE
fix(react-card): allow `CardPreview` to render under the `Card` border

### PR DESCRIPTION
**Pull request by andrefcdias**
## Current Behavior

Due to the usage of `overflow: hidden`, the image doesn't fully overflow the parent and will not grow farther than the border, even though the border is transparent.

image

## New Behavior

Given that we shouldn't show the content over the border, as high contrast themes show borders regardless of transparency, I've moved the border into the `::after` pseudo-element so that the border can render over the content.
This way, there's no 1px spacing on light/dark themes and the image doesn't cover the border on high contrast themes: 

image
image

## Related Issue(s)

#19336

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d36d823ae024c36d13c1b41764ac5095e7fabe44'>d36d823ae024c36d13c1b41764ac5095e7fabe44</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/Avatar-mount_base.html">841</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/Avatar-mount.html">926</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/Button-mount_base.html">564</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/Button-mount.html">532</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/FluentProvider-mount_base.html">1769</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/FluentProvider-mount.html">1843</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/FluentProviderWithTheme-mount_base.html">296</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/FluentProviderWithTheme-mount.html">285</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">229</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">243</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">330</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">301</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/MakeStyles-mount_base.html">1559</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22761/perf-test-react-components/MakeStyles-mount.html">1509</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by andrefcdias — COMMENTED** [review]
Screener changes expected, will approve once the PR gets approved

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (Hotell) [inline packages/react-components/react-card/src/components/Card/useCardStyles.ts:63]: **thought:** wondering if this "reset" back to default state is necessary for pseudo selectors state snapshot. 

maybe we can use something like  `...shorthands.borderColor('inherit'),` ?

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (Hotell) [inline packages/react-components/react-card/src/components/Card/useCardStyles.ts:58]: out of topic for this PR, but what about `:focused`  ? also all hover states should be probably turned off on touch devices, do we have any guidance on that from design ?

**Review by Hotell — COMMENTED** [review]
Added some questions and suggestions. Last but not least what about vr-tests ? this starts to contain a lot of UI rendering logic.

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (andrefcdias) [reply packages/react-components/react-card/src/components/Card/useCardStyles.ts:39]: These are here to be addressed with the `size` prop as they shouldn't be "static". I'll add a TODO to make it clearer.

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (andrefcdias) [reply packages/react-components/react-card/src/components/Card/useCardStyles.ts:55]: In retrospect, I can add a style that has this:
```
//pending style name™️
interactiveNoOutline: {
  ':hover::after': {
    ...shorthands.borderColor(tokens.colorTransparentStrokeInteractive),
  },
  ':active::after': {
    ...shorthands.borderColor(tokens.colorTransparentStrokeInteractive),
  },
},
```
This would be applied to all interactive appearances, except for `outline` as this uses different colors.

If you mean that just to reduce duplications of `...shorthands.borderColor(tokens.colorTransparentStrokeInteractive)`, I guess we could extract it into a variable, but the styles would also become less readable IMO.

To clarify, my fear is the detachment of what's happening becoming so great that the maintenance of this code becomes a big burden. Having pseudo elements for borders already puts such a complexity to our styles that I feel "optimizing" would create even more entropy when understanding what's going on.

WDYT about these approaches? Let me know if you have other suggestions too

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (andrefcdias) [reply packages/react-components/react-card/src/components/Card/useCardStyles.ts:63]: This isn't a reset to the default state as the default is the token `colorTransparentStroke`.

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (andrefcdias) [reply packages/react-components/react-card/src/components/Card/useCardStyles.ts:58]: Focus indicators have yet to be tackled. The goal is to use indicators as all other components do.
Regarding touch devices, I haven't yet read anything explicitly about it.

**Comment by andrefcdias** [comment]
> Added some questions and suggestions. Last but not least what about vr-tests ? this starts to contain a lot of UI rendering logic.

That is the reason why vr-tests are failing. Do you mean adding more cases? One of my goals in the end is to implement all the Card templates prototyped by design but I feel it doesn't make sense to put all the effort in now as the component is not yet finished.

**Comment by Hotell** [comment]
> That is the reason why vr-tests are failing. 

ah so this behaviour is already covered in tests ? if yes then no need to add additional ones as you mentioned

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (Hotell) [reply packages/react-components/react-card/src/components/Card/useCardStyles.ts:55]: > In retrospect, I can add a style that has this:

that would be definitely more preferable then the current approach IMO

> If you mean that just to reduce duplications of ...shorthands.borderColor(tokens.colorTransparentStrokeInteractive), I guess we could extract it into a variable, but the styles would also become less readable IMO.

No I meant if there is some API to create custom mixins (like one can do in [sass](https://sass-lang.com/documentation/at-rules/mixin) or [vanilla-extract keyframes](https://vanilla-extract.style/documentation/styling-api/#keyframes) etc)

> Having pseudo elements for borders already puts such a complexity to our styles that I feel "optimizing" would create even more entropy when understanding what's going on.

👍

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (Hotell) [reply packages/react-components/react-card/src/components/Card/useCardStyles.ts:58]: k, ty

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (Hotell) [inline packages/react-components/react-card/src/components/Card/useCardStyles.ts:39]:
```suggestion
      // TODO: Extract this to separate styles when tackling the `size` prop
```

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (Hotell) [inline packages/react-components/react-card/src/components/Card/useCardStyles.ts:44]:
```suggestion
      // TODO: Extract this to separate styles when tackling the `size` prop
```

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (Hotell) [reply packages/react-components/react-card/src/components/Card/useCardStyles.ts:63]: got it, the whole style declaration is bit confusing that what got me wondering I guess. ty

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (andrefcdias) [reply packages/react-components/react-card/src/components/Card/useCardStyles.ts:55]: > that would be definitely more preferable than the current approach IMO

I'll apply the changes then, let's see how it turns out in practice.

> No I meant if there is some API to create custom mixins (like one can do in [sass](https://sass-lang.com/documentation/at-rules/mixin) or [vanilla-extract keyframes](https://vanilla-extract.style/documentation/styling-api/#keyframes) etc)

Not that I'm aware of unfortunately. 😞

Comment (andrefcdias) [reply packages/react-components/react-card/src/components/Card/useCardStyles.ts:64]: I'm personally not a fan of how it looks. I think I'll need to do some refactoring of the styles for it to look more readable but that's out of this scope.
Let me know if you think I should leave it as is or revert the last commit @Hotell.

**Review by Hotell — APPROVED** [review]

File: packages/react-components/react-card/src/components/Card/useCardStyles.ts
Comment (Hotell) [reply packages/react-components/react-card/src/components/Card/useCardStyles.ts:64]: lets move forward with lats change. ty

**Pull request merged**
